src/global_logic.js

```diff
@@ -680,16 +691,90 @@ module.exports.calcBasedOneSummon = function (summonind, prof, buff, totals) {
+        else if (totals[key]['covenant'] === 'deleterious') {
+            var hasCrit = false;
+            var critRate = 0;
+            for (var val in criticalArray) {
+              hasCrit = true;
+              break;
+            }
+            if (hasCrit && !isNaN(criticalArray[1])) {
+              critRate = 1.0 - criticalArray[1];
+            } else if (hasCrit) {
+              critRate = 1.0;
+            }
+            if (critRate != 0) {
+              supplementalDamageArray['deleterious'] = {};
+              supplementalDamageArray['deleterious']['damage'] = Math.ceil(critRate * 30000);
+              supplementalDamageArray['deleterious']['ougiDamage'] = Math.ceil(critRate * 30000);
+              supplementalDamageArray['deleterious']['damageWithoutCritical'] = 0;
+            }
+        }
+        else if (totals[key]['covenant'] === 'calamitous') {
+            supplementalDamageArray['calamitous'] = {};
+            var debuffsCount = Math.min(10, totals['Djeeta']['debuffsCount']);
+            supplementalDamageArray['calamitous']['damage'] = debuffsCount * 3000;
+            supplementalDamageArray['calamitous']['ougiDamage'] = debuffsCount * 3000;
+            supplementalDamageArray['calamitous']['damageWithoutCritical'] = debuffsCount * 3000;
+        }
+        
+        for (var supplementalDamageKey in supplementalDamageArray) {
+            if (supplementalDamageKey != 'contentious') {
+                damage += supplementalDamageArray[supplementalDamageKey].damage;
+                ougiDamage += supplementalDamageArray[supplementalDamageKey].ougiDamage;
+                damageWithoutCritical += supplementalDamageArray[supplementalDamageKey].damageWithoutCritical;
+            }
         }
+        
+        if (expectedTurn === Infinity) {
+            var expectedCycleDamagePerTurn = expectedAttack * damage;
+
+            if (supplementalDamageArray['contentious'] != undefined) {
+                expectedCycleDamagePerTurn += supplementalDamageArray.contentious.damage;
+                damage += supplementalDamageArray.contentious.damage;
+                damageWithoutCritical += supplementalDamageArray.contentious.damageWithoutCritical;
+            }
+        } else {
+            // Normal attack * n times
+            var expectedCycleDamage = expectedTurn * expectedAttack * damage;
+            // Ougi + chain burst (buff ["chainNumber"] is 1 or more so division OK)
+            expectedCycleDamage += ougiDamage + chainBurst / buff["chainNumber"];
+
+            if (supplementalDamageArray['contentious'] != undefined) {
+                expectedCycleDamage += supplementalDamageArray.contentious.damage * expectedTurn;
+                damage += supplementalDamageArray.contentious.damage;
+                damageWithoutCritical += supplementalDamageArray.contentious.damageWithoutCritical;
+            }
 
+            var expectedCycleDamagePerTurn = expectedCycleDamage / (expectedTurn + 1.0);
+        }
```

Comment:
```diff
- var expectedCycleDamage = expectedTurn * expectedAttack * damage;
+ let expectedCycleDamage = expectedTurn * expectedAttack * damage;
```

Can you change the declare keyword "var" to "let" for `expectedCycleDamage.`

Motocal used "var"  the most cases. mixing let and var without reason, may make a confuse.
However, "var" in if block make a new issue, when Infinity case expectedCycleDamage is undefined.

Change the "var" -> "let" can resolve this by expectedCycleDamage is undefined after the if-block.
we should not use the unbalanced declared variable after the scope gone.

`expectedCycleDamagePerTurn` must use "var" if it's declared in the if-block.  this is ok.

----
Another solution is make a function to calc the expectedCycleDamagePerTurn.
As I remember, there was two places which calced expectedTurn ... in `generateHaisuiData()` function.

Reply:
currently, `expectedCycleDamagePerTurn` is being calculated once in 'normal' combination, `calcBasedOneSummon`, and another time (well... times) in `generateHaisuiData`. i don't think this really warrant creating a function for them. Although, this could reduce code clutter by a 'little'.